pkg/operator/controller/ingress/load_balancer_service.go

```diff
@@ -793,8 +793,8 @@ func loadBalancerServiceIsProgressing(ic *operatorv1.IngressController, service
 	}
 	if wantScope != haveScope {
 		err := fmt.Errorf("The IngressController scope was changed from %q to %q.", haveScope, wantScope)
-		switch platform.Type {
-		case configv1.AWSPlatformType, configv1.IBMCloudPlatformType:
+		_, platformHasMutableScope := platformsWithMutableScope[platform.Type]
+		if !platformHasMutableScope {
```

Comment:
Since you don't need `platformHasMutableScope` outside the `if` block, the following is a little more concise and idiomatic:
```suggestion
		if _, ok := platformsWithMutableScope[platform.Type]; !ok {
```
But if you prefer to keep the longer variable name or to keep the declaration outside the `if` block, that's fine.

Reply:
Sure, done.